docs/decisions/0001-lti-extensions-plugin.rst

```diff
@@ -0,0 +1,115 @@
+Platform wide LTI support
+----------------------------
+
+Status
+======
+
+Provisional (in discussion)
+
+Context
+=======
+
+The LTI Consumer implementation in the Open edX platform is currently limited to LTI launches inside XBlocks.
+
+The goal of this ADR is to:
+
+#. Define an architecture to enable LTI launch and configuration for multiple contexts (from block level to platform level).
+#. Make LTI configuration reusable throughout a context.
+#. Support both LTI 1.1 and LTI 1.3, as well as LTI Advantage.
+
+Decisions
+=========
+
+Develop LTI related code to be reusable in the platform context, and allow LTI to be configured for different contexts in the platform.
+This requires adding a separation layer between the framework presenting the content and the LTI logic, configuration and storage layer.
+
+The first steps were already taken to move towards this implementation with these PRs:
+
+`Merge LTI 1.3 support to master`_: Implemented support for LTI 1.3 by creating a framework agnostic python module that handles the
+LTI specific logic (such as launch message creation, token verification and OIDC flows) and passing responses as JSON objects to be
+interpreted and presented by the XBlock LTI consumer.
+The module is easily reusable in contexts outside the XBlock and all the code in `lti_consumer/lti_1p3` don't rely on XBlock or
+Django specific features.
+The PR uses the XBlock's fields to store LTI configuration.
+
+`Add support for LTI embeds in course tabs and elsewhere`_: This is a partial implementation of logic separation for the LTI 1.1
+consumer to allow LTI launches from outside the context of an XBlock, following the guidelines set by the LTI 1.3 implementation.
+While this is a smaller step than the implementation above, it allows LTI 1.1 launches on any LMS/Studio view, but doesn't implement
+the configuration storage.
+
+.. _`Merge LTI 1.3 support to master`: https://github.com/edx/xblock-lti-consumer/pull/82
+.. _`Add support for LTI embeds in course tabs and elsewhere`: https://github.com/edx/xblock-lti-consumer/pull/77
+
+
+A few actions that need to be taken to achieve the desired architecture are outlined below:
+
+Centralized configuration for LTI launches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use Django models to store LTI related configuration and launch data for all LTI integrations.
```

Comment:
I'm assuming so, but would this be used for existing LTI XBlocks as well?  i.e., when you say _all_ do you really mean _all_? 😄

Reply:
Ahh - I see this is covered below in "Tech Debt" 👍

Reply:
>I'm assuming so, but would this be used for existing LTI XBlocks as well? 

For new XBlocks, we can give the user the option to opt for the new storage or keep using the old one. And then, over the next few releases, deprecate and disable the old configuration method.

Reply:
I think the old method can be marked as deprecated already, with clear links to documentation or the new UI to add these settings. If we already know it's on the way out, marking it as deprecated sooner rather than later is better right? If need be it can just be kept around for longer.

Reply:
Yup, we should mark it as deprecated as soon as we have a configuration model working with full capabilities (models + UI + working launches).

>If we already know it's on the way out, marking it as deprecated sooner rather than later is better right?

:+1: Yes, definitively.